webrender/src/device/gfx.rs

```diff
@@ -2727,6 +2740,132 @@ impl<B: hal::Backend> Device<B> {
+            ),
+            stencil_ops: hal::pass::AttachmentOps::DONT_CARE,
+            layouts: hal::image::Layout::ColorAttachmentOptimal
+                .. hal::image::Layout::ColorAttachmentOptimal,
+        };
+
+        let attachment_depth = hal::pass::Attachment {
+            format: Some(depth_format),
+            samples: 1,
+            ops: hal::pass::AttachmentOps::new(
+                hal::pass::AttachmentLoadOp::DontCare,
+                hal::pass::AttachmentStoreOp::Store,
+            ),
+            stencil_ops: hal::pass::AttachmentOps::DONT_CARE,
+            layouts: hal::image::Layout::DepthStencilAttachmentOptimal
+                .. hal::image::Layout::DepthStencilAttachmentOptimal,
+        };
+
+        let subpass_r8 = hal::pass::SubpassDesc {
+            colors: &[(0, hal::image::Layout::ColorAttachmentOptimal)],
+            depth_stencil: None,
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let subpass_depth_r8 = hal::pass::SubpassDesc {
+            colors: &[(0, hal::image::Layout::ColorAttachmentOptimal)],
+            depth_stencil: Some(&(1, hal::image::Layout::DepthStencilAttachmentOptimal)),
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let subpass_bgra8 = hal::pass::SubpassDesc {
+            colors: &[(0, hal::image::Layout::ColorAttachmentOptimal)],
+            depth_stencil: None,
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let subpass_depth_bgra8 = hal::pass::SubpassDesc {
+            colors: &[(0, hal::image::Layout::ColorAttachmentOptimal)],
+            depth_stencil: Some(&(1, hal::image::Layout::DepthStencilAttachmentOptimal)),
+            inputs: &[],
+            resolves: &[],
+            preserves: &[],
+        };
+
+        let dependency = hal::pass::SubpassDependency {
+            passes: hal::pass::SubpassRef::External .. hal::pass::SubpassRef::Pass(0),
+            stages: PipelineStage::COLOR_ATTACHMENT_OUTPUT
+                .. PipelineStage::COLOR_ATTACHMENT_OUTPUT,
+            accesses: hal::image::Access::empty()
+                .. (hal::image::Access::COLOR_ATTACHMENT_READ
+                    | hal::image::Access::COLOR_ATTACHMENT_WRITE),
+        };
+
+        let depth_dependency = hal::pass::SubpassDependency {
+            passes: hal::pass::SubpassRef::External .. hal::pass::SubpassRef::Pass(0),
+            stages: PipelineStage::EARLY_FRAGMENT_TESTS .. PipelineStage::LATE_FRAGMENT_TESTS,
+            accesses: hal::image::Access::empty()
+                .. (hal::image::Access::DEPTH_STENCIL_ATTACHMENT_READ
+                    | hal::image::Access::DEPTH_STENCIL_ATTACHMENT_WRITE),
+        };
+
+        RenderPass {
+            r8: unsafe {
+                device.create_render_pass(
+                    &[attachment_r8.clone()],
```

Comment:
super nit: can we do `&[&attachment_r8]` or `iter::once(&attachment_r8)`?

Reply:
Yes, we can change it to `iter::once`.